src/Common/src/Certificates/ConfigureCertificateOptions.cs

```diff
@@ -42,16 +47,29 @@ public void Configure(string? name, CertificateOptions options)
 
         string? privateKeyFilePath = _configuration.GetValue<string>(GetConfigurationKey(name, "PrivateKeyFilePath"));
 
-        options.Certificate = privateKeyFilePath != null && File.Exists(privateKeyFilePath)
-            ? X509Certificate2.CreateFromPemFile(certificateFilePath, privateKeyFilePath)
-            : new X509Certificate2(certificateFilePath);
+        try
+        {
+            options.Certificate = privateKeyFilePath != null && File.Exists(privateKeyFilePath)
+                ? X509Certificate2.CreateFromPemFile(certificateFilePath, privateKeyFilePath)
+                : new X509Certificate2(certificateFilePath);
 
-        X509Certificate2[] certificateChain = CertificateRegex.Matches(File.ReadAllText(certificateFilePath))
-            .Select(x => new X509Certificate2(Encoding.ASCII.GetBytes(x.Value))).ToArray();
+            X509Certificate2[] certificateChain = CertificateRegex.Matches(File.ReadAllText(certificateFilePath))
+                .Select(x => new X509Certificate2(Encoding.ASCII.GetBytes(x.Value))).ToArray();
 
-        foreach (X509Certificate2 issuer in certificateChain.Skip(1))
+            foreach (X509Certificate2 issuer in certificateChain.Skip(1))
+            {
+                options.IssuerChain.Add(issuer);
+            }
+        }
+        catch (IOException ex)
+        {
+            _logger.LogDebug(ex, "IOException while loading certificate for '{CertificateName}' from '{Path}'. Will retry on next reload.", name,
+                certificateFilePath);
+        }
+        catch (CryptographicException ex)
         {
-            options.IssuerChain.Add(issuer);
+            _logger.LogDebug(ex, "CryptographicException while parsing certificate for '{CertificateName}' from '{Path}'. Will retry on next reload.", name,
```

Comment:
Should this log at warning level? If an invalid certificate is deployed, an operator likely wants to know. But if this is a transient failure that auto-fixes itself within the next second, logging at Debug level would be appropriate.

Reply:
If we combine the exception handling, we'll lose the ability to differentiate between the two types. I've only seen `IOException` with the transient failures and only seen `CryptographicException` with the bad file content.